### ISSUE TITLE
Annotate result columns with "breakout: true" or "aggregation: true" when possible

### ISSUE THREAD

**Issue by tlrobinson** [metabase/metabase#2729]
For structured queries the query processor should annotate the data.cols objects with boolean flags indicating whether they are breakouts or aggregations to allow us to disambiguate numeric dimensions and metrics when figuring out how to display results as a chart.

e.x. 

```javascript
{
  data: {
    cols: [
      {
        name: "CREATED_AT",
        ...
        dimension: true
      },
      {
        name: "count",
        ...
        aggregation: true
      }
    ],
    ...
} 
```

Necessary for #2549

**Comment by salsakran** [comment]
Is there anything else we can add to the result dictionary that could be used by the front end to do Smart Things?

**Comment by salsakran** [comment]
And do we want to return this as a flag in the column or as a separate data structure? Eg, a mapping of the original query to `cols` indexes?

**Comment by camsaul** [comment]
Why would we want a separate data structure?

**Comment by salsakran** [comment]
just asking =)

It might be useful in other parts of front end. Don't want to be too myopic. While narrowly spec'd, it's a fix for #2730, I don't want to overly specialize the return format and just accumulate layers of bandaids.

**Comment by tlrobinson** [comment]
I think a flag on the column makes sense, however, I could see doing something like `breakout: 0` or `breakout: 1` depending on the position of the breakout, rather than a simple boolean. It's not necessary for #2730 but could be useful in the future.

**Comment by camsaul** [comment]
I think we should table consideration of adding all these extra things for now seeing how code complete is a day away. I want to keep the scope of this issue limited to what's needed so I can deliver it time

**Comment by tlrobinson** [comment]
That's fine, but *if* it's trivial to use the index of the breakout rather than `true` (I'm not saying it is) then I'd say go ahead and do it.

We can use a similar pattern for #2730

**Comment by salsakran** [comment]
Yeah, my position here is that if it's a question of including things in a return data structure that are accessible in the context that structure will be created in, we can add them if they will be useful. 

Things that are not accessible and will require significant work are out of scope.

**Comment by agilliland** [comment]
I agree with Tom that it makes the most sense for these to just be considered attributes of the column, so I would just add them to what we have there already.

The idea of using the position of the breakout is neat and if its easy to do then go for it, but realistically that shouldn't be needed because it seems pretty safe to just use the column ordering from the result.  

In general I think that query results should be things that stand on their own as much as possible and we should avoid linking them too heavily with the original query they are based on, partly so avoid over entwining queries and results and partly so that native queries don't drift too far into a no-mans-land of broken capabilities.

**Comment by camsaul** [comment]
Implemented by #2731
